pkg/kubelet/status/status_manager.go

```diff
@@ -713,3 +713,41 @@ func NeedToReconcilePodReadiness(pod *v1.Pod) bool {
 	}
 	return false
 }
+
+//SidecarsStatus contains three bools, whether the pod has sidecars, if the all the sidecars are ready and if the non sidecars are in a waiting state.
+type SidecarsStatus struct {
+	SidecarsPresent   bool
+	SidecarsReady     bool
+	ContainersWaiting bool
+}
+
+//GetSidecarsStatus returns the SidecarsStatus for the given pod
+func GetSidecarsStatus(pod *v1.Pod) SidecarsStatus {
+
+	if pod == nil {
+		return SidecarsStatus{SidecarsPresent: false, SidecarsReady: false, ContainersWaiting: false}
+	}
+	if pod.Spec.Containers == nil || pod.Status.ContainerStatuses == nil {
+		return SidecarsStatus{SidecarsPresent: false, SidecarsReady: false, ContainersWaiting: false}
+	}
+
+	sidecarsStatus := SidecarsStatus{SidecarsPresent: false, SidecarsReady: true, ContainersWaiting: false}
+
+	for _, container := range pod.Spec.Containers {
+		for _, status := range pod.Status.ContainerStatuses {
```

Comment:
following up my comment in the api change PR, if we can make container lifecycle as part of `corev1.ContainerStatus`, that could possibly make a lot of operations easier. This is one example here.

Reply:
I agree that we could make this nicer, but currently this 'works' and I'm not sure if we should spread this alpha feature over more parts of the API. Be interested in what others think.